extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -2117,7 +2117,7 @@ private void updateLatestOffsetsFromKafka()
                      && latestOffsetsFromKafka.get(e.getKey()) != null
                      && e.getValue() != null
                      ? latestOffsetsFromKafka.get(e.getKey()) - e.getValue()
-                     : null
+                     : Integer.MIN_VALUE
```

Comment:
Maybe is the below better because we don't have to add unnecessary lag values?

```java
private Map<Integer, Long> getLagPerPartition(Map<Integer, Long> currentOffsets)
  {
    if (latestOffsetsFromKafka == null) {
      return ImmutableMap.of();
    }

    return currentOffsets
        .entrySet()
        .stream()
        .filter(e -> latestOffsetsFromKafka.get(e.getKey()) != null && e.getValue() != null)
        .collect(
            Collectors.toMap(
                Map.Entry::getKey,
                e -> latestOffsetsFromKafka.get(e.getKey()) - e.getValue()
            )
        );
  }
```

Reply:
I didn't wanted to filter so that its visible that there is mismatch between task count and number of available Kafka partitions. Currently, whenever total lag is calculated, `x -> Math.max(x, 0)` is used so setting lag to `Integer.MIN_VALUE` won't add to the total.

Reply:
However, if you prefer this then we can do this as well.

Reply:
Both looks good to me. If you think the current patch is better to check there is mismatch between task count and number of partitions, please go for it.